src/gsi/fca_gsi_inter.f90

```diff
@@ -0,0 +1,1101 @@
+module fca_gsi_inter_m
```

Comment:
Need clean up the code, add proper documentation. I will review again.

Reply:
ok, comments cleaned up